### ISSUE TITLE
php71u-pecl-mongodb

### ISSUE THREAD

**Issue by allella** [iuscommunity/wishlist#130]
### What package do you want?

php71u-pecl-mongodb 

I'm assuming the php70u folks would want it too, so secondarily I'll mention php70u-pecl-mongodb

### Why?

The PECL mongodb driver in [EPEL is 1.1.10](https://dl.fedoraproject.org/pub/epel/7/x86_64/p/), but the [latest available version is 1.2.8](https://pecl.php.net/package-changelog.php?package=mongodb&release=1.2.8).

The stable [MongoDB 3.4 requires PECL mongodb with PECL mongodb version 1.2+](https://docs.mongodb.com/ecosystem/drivers/php/#compatibility).

This is the [latest suggested PHP driver by MongoDB](https://docs.mongodb.com/ecosystem/drivers/php/#drivers).

Since PECL mongodb is a lighweight driver it virtually requires the the [Mongo PHP Library](https://github.com/mongodb/mongo-php-library) as a userland and that Composer library [now depends on 1.2+ in the composer.json](https://github.com/mongodb/mongo-php-library/blob/master/composer.json)

The older mongo package is marked "legacy" on both [PHP.net](http://php.net/mongo) and the [PECL mongo](https://pecl.php.net/package/mongo) page.

As [Remi mentioned](https://github.com/iuscommunity/wishlist/issues/72#issuecomment-225137767), PECL mongo will never be made for PHP 7 so the requested PECL mongodb is the present and future.

Finally, IUS has previously supplied a [Mongo PECL driver since PHP 5.3](https://github.com/iuscommunity-pkg?utf8=%E2%9C%93&q=pecl-mongo&type=&language=), so this would continue the trend (and fill a current gap) in IUS by supporting a MongoDB PECL driver for PHP 7.x

### OS versions

I'd be using it for CentOS/RHEL 7. 

Secondarily, I don't supposed RHEL/CentOS 6 would be a bad idea since [IUS supports PHP 7.0 and 7.1 for RHEL/CentOS 6](https://dl.iuscommunity.org/pub/ius/stable/CentOS/6/x86_64/).

### Testing

I don't yet have any production environments running PECL mongodb, but I'd be willing to fire up at least a CentOS 7 PHP setup and see that it installs and run through some [documentation examples](http://php.net/manual/en/mongodb.tutorial.library.php) with the userland installed to make sure it doesn't barf.

**Comment by carlwgeorge** [comment]
Thanks for the request @allella.  I'm looking over the [MongoDB PHP architecture](http://php.net/manual/en/mongodb.overview.php).  This one looks like it might be a bit hairy because of the minimum library requirements.  As of version 1.2, it requires at least libbson 1.5 and libmongoc 1.5.  Those libraries aren't available on EL6, and too old in EL7.

* https://apps.fedoraproject.org/packages/libbson
* https://apps.fedoraproject.org/packages/mongo-c-driver

I'm still researching to determine the best way to approach this.

**Comment by carlwgeorge** [comment]
Turns out those libraries come [bundled in the PECL tarball](https://github.com/mongodb/mongo-php-driver/commit/e25091b).  Normally we stick with Fedora/Red Hat [packaging standards](https://fedoraproject.org/wiki/Bundled_Libraries) and [unbundled libraries](https://src.fedoraproject.org/cgit/rpms/php-pecl-mongodb.git/tree/php-pecl-mongodb.spec?id=b23af82b39e311aa32641c6d4a55b00b97d44b3c#n87), but in this case I see little benefit.  Unbundling them would require we maintain two more packages, and nothing else uses them besides the mongodb PECL module.

I've finished the [spec file](https://github.com/iuscommunity-pkg/php71u-pecl-mongodb), and the initial packages will move to the testing repositories during tonight's automation.

**Comment by allella** [comment]
That was fast. Thanks.

I'll give it a test in the coming days and provide feedback.

**Comment by allella** [comment]
This installed and I did a basic example against MongoDB 3.0 and 3.2 without any issues.

- Installed via the test repo using `yum --enablerepo=ius-testing install php71u-pecl-mongodb`
- Reloaded PHP-FPM and checked that /etc/php.d/50-mongodb.ini was added and available to PHP.
- Installed the [Mongo PHP Library](https://github.com/mongodb/mongo-php-driver) in a test directory using `composer require mongodb/mongodb`
- Ran a [simple findOne example](https://docs.mongodb.com/php-library/master/reference/method/MongoDBCollection-findOne/), like below.

```
<?php
require 'vendor/autoload.php'; // include Composer's autoloader
$client = new MongoDB\Client("mongodb://localhost:27017");
$collection = $client->local->startup_log;
$result = $collection->findOne();
print_r($result);
```
